### PR TITLE
AWS SDK v3

### PR DESCRIPTION
**Pull request by erikzaadi**
Upgrade core and persistent pool to use the new aws sdk v3

---

_Additional context for reviewer_

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/persistent-pool/src/lib/dynamodb/dynamodb_repo.ts [hunk 1]
```diff
@@ -125,7 +143,7 @@ export const repo = defaultOpts.withRequired<
     try {
       return await f()
     } catch (e) {
-      if (e.code === 'ConditionalCheckFailedException') {
+      if (e.toString().includes('ConditionalCheckFailedException')) {
```
Comment (ori-moisis) [inline packages/persistent-pool/src/lib/dynamodb/dynamodb_repo.ts:146]: is there no structure to the error now?

File: packages/persistent-pool/src/lib/dynamodb/utils.ts [hunk 2]
```diff
@@ -37,15 +51,17 @@ export const dbUtils = (db: DynamoDB) => {
...
-      await db.createTable(tableParams).promise()
+      await db.send(new CreateTableCommand(tableParams))
     } catch (e) {
-      if (e.code !== 'ResourceInUseException') throw e
+      if (!e.toString().includes('ResourceInUseException')) {
+        throw e
+      }
     }
-    await withRetry(() => tableExists(tableParams.TableName), waitOpts)
+    await withRetry(() => tableExists(tableParams.TableName ?? ''), waitOpts)
```
Comment (ori-moisis) [inline packages/persistent-pool/src/lib/dynamodb/utils.ts:64]: should we maybe require `TableName` to be here?
feels like checking for a table named `''` could potentially be a problem (and I assume in reality this code path is never taken, right?)

**Review by ori-moisis — APPROVED** [review]
image

File: packages/persistent-pool/src/lib/dynamodb/dynamodb_repo.ts (hunk 1 above)
Comment (erikzaadi) [reply packages/persistent-pool/src/lib/dynamodb/dynamodb_repo.ts:146]: Nope, only has a `$metadata` key (when I inspected it)

**Pull request merged**